default/tests/shared/database/source_control/test_working_dir.py

```diff
@@ -66,32 +68,102 @@ def test_list(self):
             'access_type': 'restricted'
         }, self.session)
 
+        mock_perm_result = Mock()
+        mock_perm_result.allow_all = False
+        mock_perm_result.allowed_object_id_list = [ds1.id, ds2.id, ds3.id]
+        mock_get_dataset_viewing_permissions.return_value = mock_perm_result
+
         dirs = WorkingDir.list(
             session = self.session,
             project_id = self.project.id,
             member = user.member,
         )
         dirs_id_list = [x.id for x in dirs]
-        self.assertEqual(len(dirs), 6)
+        self.assertEqual(len(dirs), 3)
```

Comment:
Not sure why this was checking `6` before, but with new mocking we only get the 3 directories created above